zio-http/src/main/scala/zio/http/netty/client/NettyConnectionPool.scala

```diff
@@ -127,9 +125,24 @@ object NettyConnectionPool {
     } yield result
   }
 
+  private final class ReadTimeoutErrorHandler(nettyRuntime: NettyRuntime)(implicit trace: Trace)
+      extends ChannelInboundHandlerAdapter {
```

Comment:
Also I had an idea of implementing `ReadTimeoutHandler` itself and rewrite `readTimedOut` method, so instead of raising an exception, it would just write a log message. But I'm not sure which wy is more correct/idiomatic